src/sqlfluff/rules/L015.py

```diff
@@ -35,46 +36,81 @@ class Rule_L015(BaseRule):
-        if bracketed:
-            # If there's nothing else in the expression, remove the brackets.
-            if len(expression[0].segments) == 1:
-                # Remove the brackets and strip any meta segments.
-                anchor = bracketed.get()
-                assert anchor
-                seq = ReflowSequence.from_around_target(
-                    anchor,
-                    context.parent_stack[0],
-                    config=context.config,
-                    sides="before",
-                ).replace(anchor, self.filter_meta(anchor.segments)[1:-1])
-            # Otherwise, still make sure there's a space after the DISTINCT.
-            else:
-                anchor = modifier[0]
-                seq = ReflowSequence.from_around_target(
-                    modifier[0],
-                    context.parent_stack[0],
-                    config=context.config,
-                    sides="after",
-                )
+        if context.segment.is_type("select_clause"):
+            # Look for `select_clause_modifier`
+            modifier = children.select(sp.is_type("select_clause_modifier"))
+            first_element = children.select(sp.is_type("select_clause_element")).first()
+            expression = (
+                first_element.children(sp.is_type("expression")).first()
+                or first_element
+            )
+            bracketed = expression.children(sp.is_type("bracketed")).first()
+            # is the first element only an expression with only brackets?
+            if modifier and bracketed:
+                # If there's nothing else in the expression, remove the brackets.
+                if len(expression[0].segments) == 1:
+                    anchor, seq = self._remove_unneeded_brackets(context, bracketed)
+                # Otherwise, still make sure there's a space after the DISTINCT.
+                else:
+                    anchor = modifier[0]
+                    seq = ReflowSequence.from_around_target(
+                        modifier[0],
+                        context.parent_stack[0],
+                        config=context.config,
+                        sides="after",
+                    )
+        elif context.segment.is_type("function"):
+            # Look for a function call DISTINCT() whose parent is an expression
+            # with a single child.
+            if (
+                not context.parent_stack[-1].is_type("expression")
+                or len(context.parent_stack[-1].segments) != 1
+            ):
+                return None
+            function_name = children.select(sp.is_type("function_name")).first()
+            bracketed = children.first(sp.is_type("bracketed"))
+            if (
+                not function_name
+                or function_name[0].raw_upper != "DISTINCT"
+                or not bracketed
+            ):
+                return None
+            anchor = context.parent_stack[-1]
```

Comment:
Better to define `anchor` at line 75 and then check `anchor` in `if` on lines 76 and 77?

Reply:
Good point -- I'll change it.